openssl/src/crypto/pkey.rs

```diff
@@ -205,6 +205,27 @@ impl PKey {
         }
     }
 
+    /// pass ownership of the RSA key to this
+    pub fn set_rsa(&mut self, rsa: RSA) {
```

Comment:
Seems like we could get away with passing a `&RSA` here since `EVP_PKEY_set1_RSA` will increment the refcount.